### PR TITLE
Refactoring AuthService

### PR DESCRIPTION
**Pull request by AhmedHazim1**

### CODE REVIEW

File: Controllers/Mobile/v1/AuthController.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public async Task<IActionResult> ResendOtpAsync([FromBody] PhoneNumberReq phoneN
             if (!ModelState.IsValid)
                 return CustomBadRequest();
 
-            var storedOTP = await _authService.FindOtpAsync(phoneNumberReq.PhoneNumber);
+            var storedOTP = await _authService.GetOtpByPhoneNumber(phoneNumberReq.PhoneNumber);
```
Comment (MohammedLilo) [inline Controllers/Mobile/v1/AuthController.cs:27]: Please, prefix every method that queries the database for a record with `Find`. As this is the convention used by the rest of the contributors.

File: Models/DTOs/OtpDTO.cs
Comment (MohammedLilo) [inline Models/DTOs/OtpDTO.cs:1]: OtpInputDTO should contain only the phone number and the otp code, the other properties are irrelevant in this context.

File: Services/TempUserService.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+namespace AonFreelancing.Services
+{
+    public class TempUserService : MainDbService
+    {
+        public TempUserService(MainAppContext mainAppContext) : base(mainAppContext){}
+        
+        public async Task<TempUser?> GetByIdAsync(int id) => await _mainAppContext.TempUsers.FindAsync(id);
+        public async Task<IEnumerable<TempUser>> GetAllAsync() => await _mainAppContext.TempUsers.ToListAsync();
+        public async Task<TempUser?> GetByPhoneNumberAsync(string phoneNumber) => await _mainAppContext.TempUsers.Where(tu => tu.PhoneNumber == phoneNumber).FirstOrDefaultAsync();
+        public TempUser Create(TempUserDTO tempUserDTO) => new TempUser(tempUserDTO.PhoneNumber);
```
Comment (MohammedLilo) [inline Services/TempUserService.cs:24]: this should take only the phone number as an input.

File: Services/RemoveService.cs
[the remark this answers is not in the document]
Comment (AhmedHazim1) [reply Services/RemoveService.cs:1]: What I got from your commit, that it should be removed, and get back to the autheService.
Correct?

**Review by MohammedLilo — COMMENTED** [review]
- You should not create and modify the DTOs while refactoring the auth service and its related services.
- Use the previously used DTOs as is (do not add or remove properties, but feel free to add a constructor if you want), and if you think they require refactoring,  you can schedule it to another PR.

File: Controllers/Mobile/v1/TasksController.cs [hunk 3]
```diff
@@ -2,65 +2,105 @@
...
+            return BadRequest(CreateErrorResponse(StatusCodes.Status400BadRequest.ToString(), $"task status must be { Constants.TASK_STATUS_IN_PROGRESS } to start."));
+        await taskService.SubmitTaskAsync(storedTask);
         return Ok(CreateSuccessResponse(TaskOutputDTO.FromTask(storedTask)));
     }
+
+    // Implement Task approval (only clients owner can) In review -> Done (Update CompletedAt )
     [Authorize(Roles = Constants.USER_TYPE_CLIENT)]
-    [HttpPut("{id}")]
-    public async Task<IActionResult> UpdateByIdAsync(long id, [FromBody] TaskInputDTO taskInputDTO)
+    [HttpPut("{id}/ApproveTask")]
```
Comment (diyar214) [inline Controllers/Mobile/v1/TasksController.cs:65]: Use `kebab-case`

File: Controllers/Mobile/v1/TasksController.cs [hunk 4]
```diff
@@ -2,65 +2,105 @@
...
-
+        if(storedTask.Status != Constants.TASK_STATUS_IN_REVIEW)
+            return BadRequest(CreateErrorResponse(StatusCodes.Status400BadRequest.ToString(), $"task status must be { Constants.TASK_STATUS_IN_REVIEW } to start."));
+        await taskService.ApproveTaskAsync(storedTask);
         return Ok(CreateSuccessResponse(TaskOutputDTO.FromTask(storedTask)));
     }
 
+    // rejection (only clients owner can) In review -> (In progress ? To Do) 
+    [Authorize(Roles = Constants.USER_TYPE_CLIENT)]
+    [HttpPut("{id}/RejectTask")]
```
Comment (diyar214) [inline Controllers/Mobile/v1/TasksController.cs:87]: same here

**Review by diyar214 — APPROVED** [review]
LGTM!

**Pull request merged**